ios/Classes/PTPluginUtils.m

```diff
@@ -20,35 +58,459 @@ + (NSNumber *)PT_idAsNSNumber:(id)value
+}
+
++ (id)PT_JSONStringToId:(NSString *)jsonString {
+    NSData *annotListData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+    return [NSJSONSerialization JSONObjectWithData:annotListData options:kNilOptions error:nil];
+}
+
++ (PTAnnot *)findAnnotWithUniqueID:(NSString *)uniqueID onPageNumber:(int)pageNumber documentViewController:(PTDocumentViewController *)docVC error:(NSError **)error
+{
+    if (uniqueID.length == 0 || pageNumber < 1) {
+        return nil;
+    }
+    PTPDFViewCtrl *pdfViewCtrl = docVC.pdfViewCtrl;
+    __block PTAnnot *resultAnnot;
+
+    [docVC.pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+        NSArray<PTAnnot *> *annots = [pdfViewCtrl GetAnnotationsOnPage:pageNumber];
+        for (PTAnnot *annot in annots) {
+            if (![annot IsValid]) {
+                continue;
+            }
+            
+            // Check if the annot's unique ID matches.
+            NSString *annotUniqueId = nil;
+            PTObj *annotUniqueIdObj = [annot GetUniqueID];
+            if ([annotUniqueIdObj IsValid]) {
+                annotUniqueId = [annotUniqueIdObj GetAsPDFText];
+            }
+            if (annotUniqueId && [annotUniqueId isEqualToString:uniqueID]) {
+                resultAnnot = annot;
+                break;
+            }
+        }
+    } error:error];
+   
+    if(*error)
+    {
+        NSLog(@"Error: There was an error while trying to find annotation with id and page number. %@", (*error).localizedDescription);
+    }
+    
+    return resultAnnot;
+}
+
++(NSArray<PTAnnot *> *)getAnnotationsOnPage:(int)pageNumber documentViewController:(PTDocumentViewController *)docVC
+{
+    __block NSArray<PTAnnot *> *annots;
+    NSError* error;
+    [docVC.pdfViewCtrl DocLockReadWithBlock:^(PTPDFDoc * _Nullable doc) {
+        annots = [docVC.pdfViewCtrl GetAnnotationsOnPage:pageNumber];
+    } error:&error];
+    
+    if (error) {
+        NSLog(@"Error: There was an error while trying to find annotations in page number. %@", error.localizedDescription);
+    }
+    
+    return annots;
+}
+
++(NSArray<PTAnnot *> *)findAnnotsWithUniqueIDs:(NSArray <NSDictionary *>*)idPageNumberPairs documentViewController:(PTDocumentViewController *)docVC error:(NSError **)error
+{
+    NSMutableArray<PTAnnot *> *resultAnnots = [[NSMutableArray alloc] init];
+    
+    NSMutableDictionary <NSNumber *, NSMutableArray <NSString *> *> *pageNumberAnnotDict = [[NSMutableDictionary alloc] init];
+    
+    // put all annotations in a dict indexed by page number
+    for (NSDictionary *idPageNumberPair in idPageNumberPairs) {
+        NSNumber *pageNumber = [PTPluginUtils PT_idAsNSNumber:idPageNumberPair[PTAnnotPageNumberKey]];
+        NSString *annotId = [PTPluginUtils PT_idAsNSString:idPageNumberPair[PTAnnotIdKey]];
+        NSMutableArray <NSString *> *annotArray;
+        if (![pageNumberAnnotDict objectForKey:pageNumber]) {
```

Comment:
Instead of using `objectForKey:`, you can use subscripting and access it more simply by writing `pageNumberAnnotDict[pageNumber]` as you've done below.